src/component.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Base Component class
+ * @param {object} props The initial component props
+ */
+function Component(props) {
+  this.props = props
+  this.state = this.state || {}
+}
+
+Component.prototype.render = function() {}
```

Comment:
fyi react doesn't provide a default `render` on `Component`: https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js

Reply:
Interesting, I see in the React docs it states a render method must be provided.

Preact _does_ include a render method (on which this code was based) https://github.com/developit/preact/blob/master/src/component.js